### ISSUE TITLE
C# client does not throw errors / impossible to get API response code

### ISSUE THREAD

**Issue by denious** [svix/svix-webhooks#823]
### Version

C# Svix NuGet package 0.77.0

### Platform

NET Framework 4.8

### Description

The C# client does not throw exceptions and it's impossible to extract the API response code from the method's return value, which is simply `null` when an exception occurs.

I tried this code:

```csharp
const string authToken = "...";
SvixOptions options = new("https://api.svix.com");
const string applicationId = "...";

SvixClient svixClient = new(authToken, options);
var response = svixClient.Message.Create(applicationId, new MessageIn(
    eventType: "message.ready",
    payload: new
    {
        Value = "test"
    },
    eventId: Guid.NewGuid().ToString()
));
```

I tried both `Create` and `CreateAsync`, and expected to get 429s when sending 100 requests/second on a Free account, and some other 4xx when sending messages to a client application that does not exist.

Instead, nothing happened at all. No error, just a null returned by the method, which doesn't help extract the response code either.

**Comment by tasn** [comment]
Whoops, forgot to reply! We'll figure it out. It's really annoying that it doesn't behave correctly here.

Please consider making direct HTTP calls until this is fixed!

**Comment by svix-onelson** [comment]
Hi @denious! I'm having a bit of trouble getting the lib built on Linux (where I live) so I haven't yet been able to verify, but here's what I've seen looking at the source.

When you create your `SvixOptions`, it can take a 2nd parameter:

https://github.com/svix/svix-webhooks/blob/3ad11fcff6a90eb67a3d08b305726f928b6c6280/csharp/Svix/Models/SvixOptions.cs#L16

When `Throw` is set `true` in the options, an `ApiException` will be thrown instead of swallowed like you're seeing.

https://github.com/svix/svix-webhooks/blob/3ad11fcff6a90eb67a3d08b305726f928b6c6280/csharp/Svix/Message.cs#L88-L96

`ApiException` has a `ErrorCode` field which should have the HTTP status code for the failed request.

Let us know if this works for you 🙏

**Comment by tasn** [comment]
@svix-onelson, good catch. I think we should probably change the default to *YES* throw. Even though it's technically a break, it's worth doing it to have a sane default, and we'll document it in the changelog.

Can you please open a PR?

**Comment by tasn** [comment]
It's fixed in #852
